File: backend/src/utils/response.ts
import { Response } from 'express';
import { ApiResponse, PaginatedResponse } from '../types';

export function success<T>(res: Response, data: T, message?: string, statusCode = 200): Response {
  const response: ApiResponse<T> = {
    success: true,
    message,
    data,
  };
  return res.status(statusCode).json(response);
}

export function error(res: Response, message: string, statusCode = 500): Response {
  const response: ApiResponse = {
    success: false,
    error: message,
  };
  return res.status(statusCode).json(response);
}

export function paginated<T>(
  res: Response,
  data: T,
  total: number,
  page: number,
  limit: number
): Response {
  const response: PaginatedResponse<T> = {
    success: true,
    data,
    total,
    page,
    limit,
    totalPages: Math.ceil(total / limit),
  };
  return res.status(200).json(response);
}

File: backend/app/utils/response.py
import math

from fastapi.responses import JSONResponse


def success_response(data=None, message=None, status_code=200):
    body = {"success": True}
    if message:
        body["message"] = message
    if data is not None:
        body["data"] = data
    return JSONResponse(content=body, status_code=status_code)


def error_response(message="Internal server error", status_code=500):
    return JSONResponse(
        content={"success": False, "error": message}, status_code=status_code
    )


def paginated_response(data, total, page, limit):
    return JSONResponse(
        content={
            "success": True,
            "data": data,
            "total": total,
            "page": page,
            "limit": limit,
            "totalPages": math.ceil(total / limit) if limit > 0 else 0,
        }
    )
